src/legacy/server/config/transform_deprecations.js

```diff
@@ -57,6 +57,38 @@ const dataPath = (settings, log) => {
   }
 };
 
+const NONCE_STRING = `{nonce}`;
+// Policies that should include the 'self' source
+const SELF_POLICIES = Object.freeze(['script-src', 'style-src']);
+const SELF_STRING = `'self'`;
+
+const cspRules = (settings, log) => {
+  const rules = _.get(settings, 'csp.rules');
+  if (!rules) {
+    return;
+  }
+
+  const parsed = new Map(rules.map(ruleStr => {
+    const parts = ruleStr.split(/\s+/);
+    return [parts[0], parts.slice(1)];
+  }));
+
+  settings.csp.rules = [...parsed].map(([policy, sourceList]) => {
+    const sourcesWithNonce = sourceList.find(source => source.includes(NONCE_STRING));
+    if (sourcesWithNonce) {
+      log(`csp.rules no longer supports the {nonce} syntax`);
+      sourceList = sourceList.filter(source => !source.includes(NONCE_STRING));
+    }
+
+    if (SELF_POLICIES.includes(policy) && !sourceList.find(source => source.includes(SELF_STRING))) {
```

Comment:
If someone happened to set the following, we'll likely be breaking this install `csp.rules: ["default-src 'unsafe-eval' 'nonce-{nonce}'"]`. What if instead whenever we removed a nonce source, we added the 'self' source?

Reply:
That should work. I think we should still also add it if `script-src` or `style-src` don't have self or nonce.

Reply:
This seems reasonable to me.